recipes/hidapi/all/conanfile.py

```diff
@@ -0,0 +1,106 @@
+#
+# If you get an idea to solve one of this issues, please report here or fork.
+import os
+import shutil
+import glob
+from conans import ConanFile, AutoToolsBuildEnvironment, MSBuild, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class HidapiConan(ConanFile):
+    name = "hidapi"
+    description = "HIDAPI is a multi-platform library which allows an application to interface with USB and Bluetooth HID-Class devices on Windows, Linux, FreeBSD, and macOS."
+    topics = ("conan", "hidapi", "libusb")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/libusb/hidapi"
+    license = "BSD-Style"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "minosx": ['10.7', '10.8', '10.9', '10.10', '10.11', '10.12', '10.13', '10.14', '10.15', '11'],
+        "fPIC": [True, False],
+        "with_libusb": [True, False]
+    }
+    default_options = {
+        "minosx": 10.7, "fPIC": True, "with_libusb": False
+    }
+    generators = "pkg_config"
+
+    @property
+    def _source_dir(self):
+        return "hidapi-hidapi-" + self.version
+
+    def config_options(self):
+        if self.settings.os != "Macos":
+            del self.options.minosx
+        if self.settings.os != "Linux":
+            del self.options.with_libusb
+
+    def configure(self):
+        if self.settings.os == "Windows" and self.settings.arch == "x86_64":
+            raise ConanInvalidConfiguration("There's no support for x86_64 on Windows yet, please use x86")
+
+    def build_requirements(self):
+        self.build_requires("libtool/2.4.6")
+
+    def requirements(self):
+        if self.settings.os == "Linux" or self.settings.os == "FreeBSD":
+            self.requires("libusb/1.0.24")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        if self.settings.os != "Windows":
+            self.run("chmod +x ./%s/bootstrap" % self._source_dir)
+
+    def build(self):
+        if self.settings.os == "Windows":
+            if self.settings.compiler == "Visual Studio":
+                self.build_msvc()
+        else:
+            self.build_unix()
+
+    def build_msvc(self):
+        msbuild = MSBuild(self)
+        msbuild.build("%s/windows/hidapi.sln" % self._source_dir,
+                      platforms={"x86": "Win32"})
+
+    def build_unix(self):
+        self.run(os.path.join(".", "bootstrap"), cwd=self._source_dir)
+        if self.settings.os == "Macos":
+            configure = os.path.join(self._source_dir, "configure")
+            tools.replace_in_file(configure, r"-install_name \$rpath/",
+                                  "-install_name ")
+        autotools = AutoToolsBuildEnvironment(self)
+        if self.settings.os == "Macos":
+            autotools.flags.append('-mmacosx-version-min=%s' %
+                                   self.options.minosx)
```

Comment:
```suggestion
                                   self.settings.os.version)
```

Reply:
According to my verification, `settings.os.version` is applied to hidapi automatically (I guess is handled by autotools automatically).